src/storage.ts

```diff
@@ -666,7 +666,7 @@ export class Storage extends Service {
     options = Object.assign({}, options, {apiEndpoint});
 
     // Note: EMULATOR_HOST is an experimental configuration variable. Use apiEndpoint instead.
-    const baseUrl = EMULATOR_HOST || `${options.apiEndpoint}/storage/v1`;
+    const baseUrl = (EMULATOR_HOST || options.apiEndpoint) + '/storage/v1';
```

Comment:
```suggestion
    const baseUrl = new URL('/storage/v1', EMULATOR_HOST || options.apiEndpoint).href;
```

Reply:
Hmm, I think this broke the tests, looking for a better way...

Reply:
Couple things going wrong when this suggestion is applied:
1. Passing `/storage/v1` as the first arg discards any path info from the input, which the tests assert must be retained
2. This maybe could be improved by passing `./storage/v1` instead, but that still doesn't work due to the calls to `Storage.sanitizeEndpoint` above, which trim any trailing slash, which makes that relative path resolution still not work right.
3. The tests assert that a URL w/o a protocol should have the default `https` protocol inferred, but that doesn't work when passing through the `new URL()` constructor, because it reports the base url is invalid

I think I have a simple fix for this, but it runs into a weird issue where some tests are asserting odd mixes of what the test title says, and the _opposite_ of what it says.

I've pushed a change that applies that "simple fix" and makes the tests more internally consistent, but I wonder if that inconsistency was somehow intentional for some pre-existing use cases? I've left some comments in the code on the odd bits.

Reply:
Thanks for looking into this